artemis-commons/src/main/java/org/apache/activemq/artemis/utils/pools/MpscPool.java

```diff
@@ -33,12 +33,17 @@
  */
 public class MpscPool<T> extends Pool<T> {
 
+   private static final Logger logger = Logger.getLogger(MpscPool.class);
+
    public MpscPool(int maxSize, Consumer<T> cleaner, Supplier<T> supplier) {
       super(maxSize, cleaner, supplier);
    }
 
    @Override
    protected Queue<T> createQueue(int maxSize) {
+      if (logger.isTraceEnabled()) {
+         logger.trace("Creating Pool with size=" + maxSize + " for classes for " + supplier.get().getClass());
```

Comment:
this is trace, when the pool was created.


I did need this when I was debugging... at this point it's moot, but I thought it would be nice to leave what I needed on dev for future debugging.

this is log.trace.. I don't really care if it's burning a value if it's giving me good information on debugging, but I will remove it as you seem to care.

Reply:
Yep its not necessarily a huge issue, just felt a bit off exposing a field just for this. Could always add type into a constructor for example, or have the calling code log its creating a pool for Foo. Minimising differences in behaviour when debugging can be good also.